zip-0252.rst

```diff
@@ -2,7 +2,189 @@
-  Discussions-To: <https://github.com/zcash/zips/issues/397>
+  Created: 2021-02-23
+  License: MIT
+
+
+Terminology
+===========
+
+The key words "MUST", "MUST NOT", "SHOULD", and "MAY" in this document are to be
+interpreted as described in RFC 2119. [#RFC2119]_
+
+The term "network upgrade" in this document is to be interpreted as described in
+ZIP 200. [#zip-0200]_
+
+The terms "Testnet" and "Mainnet" are to be interpreted as described in
+section 3.11 of the Zcash Protocol Specification [#protocol-networks]_.
+
+**TODO: Choose code name**
+The code-name for the sixth Zcash network upgrade has not been chosen yet.
+
+
+Abstract
+========
+
+This proposal defines the deployment of the NU5 network upgrade.
+
+
+Specification
+=============
+
+NU5 deployment
+-----------------
+
+**TODO: Select/Confirm ZIPs**
+
+The primary sources of information about NU5 consensus protocol changes are:
+
+- The Zcash Protocol Specification [#protocol]_
+- ZIP 200: Network Upgrade Mechanism [#zip-0200]_
+- ZIP 216: Require Canonical Point Encodings [#zip-0216]_ *proposed*
+- ZIP 224: Orchard Shielded Protocol [#zip-0224]_ *proposed*
+- ZIP 225: Version 5 Transaction Format [#zip-0225]_ *proposed*
+- ZIP 244: Transaction Identifier Non-Malleability [#zip-0244]_ *proposed*
+
+The network handshake and peer management mechanisms defined in ZIP 201 [#zip-0201]_
+also apply to this upgrade.
+
+
+The following network upgrade constants [#zip-0200]_ are defined for the NU5
+upgrade:
+
+CONSENSUS_BRANCH_ID
+  ``0xF919A198``
+
+
+ACTIVATION_HEIGHT (NU5)
+  Testnet: TODO
+
+  Mainnet: TODO
+
+
+MIN_NETWORK_PROTOCOL_VERSION (NU5)
+  Testnet: 170014
+
+  Mainnet: 170015
+
+
+For each network (testnet and mainnet), nodes compatible with NU5 activation
+on that network MUST advertise a network protocol version that is greater than
+or equal to that network's MIN_NETWORK_PROTOCOL_VERSION (NU5).
+
+For each network, pre-NU5 nodes are defined as nodes advertising a protocol
+version less than that network's MIN_NETWORK_PROTOCOL_VERSION (NU5).
+
+Once NU5 activates on testnet or mainnet, NU5 nodes SHOULD take steps to:
+
+- reject new connections from pre-NU5 nodes on that network;
+- disconnect any existing connections to pre-NU5 nodes on that network.
+
+
+Backward compatibility
+======================
+
+Prior to the network upgrade activating on each network, NU5 and pre-NU5
+nodes are compatible and can connect to each other.
+
+Once the network upgrades, even though pre-NU5 nodes can still accept the
+numerically larger protocol version used by NU5 as being valid, NU5 nodes
+will always disconnect peers using lower protocol versions.
+
+Unlike Blossom, Heartwood, and Canopy, and like Overwinter and Sapling, NU5
+defines a new transaction version. NU5 transactions are therefore in the new
+v5 format specified by [#zip-0225]_. Therefore, pre-NU5 transactions are
+structurally invalid after NU5 activation: they can not be parsed correctly
+by pre-NU5 nodes.
```

Comment:
This copied text is wrong, v4 and v5 transactions are both supported after NU5.

Should we also say something about the consensus branch ID rejecting v4 transactions created before NU5 activation? (Is that correct?)

```suggestion
defines a new transaction version. Therefore, NU5 transactions MAY be in
the new v5 format specified by [#zip-0225]_. Unlike previous transaction
version updates, the pre-NU5 v4 transaction format remains valid after
NU5 activation. Both transaction formats MUST be accepted by NU5
nodes.

v5 transactions MUST be used for:

- Orchard transfers
- non-malleable transaction IDs

v4 transactions MUST be used for:
- Sprout transfers

Transactions that don't need these features can use either format.
```

Reply:
The `v4 transactions MUST be used for Sprout Transfers` is a modification that we hadn't talked about before, but I think that this could be a good idea; we could entirely remove Sprout support from the V5 transaction format in that case.

Reply:
Sounds good, how do we even make decisions like this? 🤷

Reply:
@teor2345 I've added this as a possible alternative to ZIP 225 in https://github.com/zcash/zips/pull/453

Reply:
> Sounds good, how do we even make decisions like this? 🤷

I'll make sure we discuss it at the Arborist call on the 11th.